Public/Src/App/Bxl/BuildXLApp.cs

```diff
@@ -1815,9 +1827,11 @@ internal bool OnConsoleCancelEvent(bool isTermination)
                 Environment.Exit(ExitCode.FromExitKind(effectiveExitKind));
             }
 #pragma warning disable ERP022 // Unobserved exception in generic exception handler
-            catch (Exception)
+            catch (Exception ex)
             {
                 // Oh my, this isn't going very well.
+                WriteErrorToConsole("Unhandled exception in exception handler");
+                WriteErrorToConsole(ex.ToString());

```

Comment:
>ToString [](start = 39, length = 8)

should we use `DemistifyToString()` here? `ToString()` might not give us useful results